src/pages/2023-12/next-js-app-router-cache/index.mdx

```diff
@@ -0,0 +1,388 @@
+}
+
+// Imagine these function calls happening in different parts of your Next.js page during SSR
+async function renderPage() {
+  // First call to fetch user data
+  const userData1 = await fetchUserData(123); // cache MISS, performs actual fetch
+
+  // Second call to fetch the same user data
+  const userData2 = await fetchUserData(123); // cache HIT, reuses result from first call
+
+  // userData1 and userData2 will be the same
+  console.log(userData1 === userData2); // true
+}
+
+renderPage();
+```
+
+In the code above, `renderPage()` makes two calls to `fetchUserData()`. The first call to `fetchUserData()` is fetched from the data cache or data store. The return value from the first call is then memoized in-memory.
+When the second call to `fetchUserData()` is made, the return value is retrieved from in-memory cache since it was memoized upon the first call, reducing redundant network calls. Once the route finishes rendering, the memoized values will be invalidated.
+
+Notice that the parameters of both requests are identical. This is crucial for memoization to work - the `url` and `options` parameter must match across all `fetch` requests.
+
+You might notice that we did not have to add any extra arguments in the `fetch` call to enable memoization. This is because memoization is enabled by default. Here are two reasons:
+
+**1. Performance Optimization:** If different components on the same page are requesting the same data, it's inefficient to fetch that data multiple times.
+
+**2. Consistency Across Components:** If data changes between requests made milliseconds apart, it could lead to inconsistencies in the rendered page. For example, two components displaying user information should show the same data if they are rendered as part of the same request.
+
+##### Opting out
+
+By default, all requests are memoized for performance. We can still opt-out, however. To do this, we simply pass the `signal` property of the `AbortController` to the `options` parameter in `fetch`.
+
+This is different to the typical use of the `AbortController`. Next.js overrides the `fetch` API and we can pass in the `signal` property to the `options` parameter, which tells Next.js that it is a new request, effectively opting out of memoization.
+
+```tsx {1, 4, 5}
+const { signal } = new AbortController();
+async function fetchUserData(userId) {
+  // The `fetch` function is automatically memoized by Next.js during SSR
+  const res = await fetch(`https://api.example.com/users/${userId}`, {
+    signal,
+  });
+  return res.json();
+}
+```
+
+The diagram below demonstrates the step-by-step process for how request memoization works.
+
+![request-memo](/articleAssets/2023-12/next-js-app-router-cache/request-memoization.png)
+
+#### Data Cache
+
+The goal of data cache is to minimize the number of calls made to our origin data source and cache the results across incoming server requests and deployments. Next.js achieves this by overriding the native `fetch` API.
```

Comment:
Again there is no explanation of the problem. This jumps straight into the solution but there is no explanation of what this cache does or why it is important.